scipy/stats/_qmc.py

```diff
@@ -383,6 +385,98 @@ def update_discrepancy(x_new, sample, initial_disc):
     return initial_disc + disc1 + disc2 + disc3
 
 
+def _perturb_discrepancy(sample, i1, i2, k, disc):
```

Comment:
This function does seem accurate, and it is faster for large designs.

Reply:
Thanks for confirming this point. I will still see if I can do some cython.

Reply:
Sounds like Pythran is the way to go now.